src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/HubMessage.cs

```diff
@@ -1,12 +1,45 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
+
 namespace Microsoft.AspNetCore.SignalR.Internal.Protocol
 {
     public abstract class HubMessage
     {
         protected HubMessage()
         {
         }
+
+        // Initialize with capacity 4 for the 2 built in protocols and 2 data encoders
+        private readonly List<SerializedMessage> _serializedMessages = new List<SerializedMessage>(4);
+
+        public byte[] WriteMessage(HubProtocolReaderWriter protocolReaderWriter)
```

Comment:
Still convinced it's going to bite us that this is on the `HubMessage` type itself.

Reply:
Having this on here is going to mess with the singleton `PingMessage.Instance`... I know you special case that elsewhere but it still makes me nervous dangit!